x-pack/plugins/security_solution/public/management/components/artifact_list_page/components/artifact_flyout.tsx

```diff
@@ -249,8 +261,28 @@ export const MaybeArtifactFlyout = memo<ArtifactFlyoutProps>(
         // Close the flyout
         // `undefined` will cause params to be dropped from url
         setUrlParams({ id: undefined, show: undefined }, true);
+
+        clearItem();
+        onSuccess();
       });
-    }, [formState.item, isEditFlow, labels, setUrlParams, submitData, toasts]);
+    }, [
+      formState.item,
+      isEditFlow,
+      labels,
+      setUrlParams,
+      submitData,
+      toasts,
+      clearItem,
+      onSuccess,
+    ]);
+
+    // set form state to match item if editing
+    useEffect(() => {
+      if (!isEditFlow) {
+        return;
+      }
+      setFormState(createFormInitialState(apiClient.listId, item));
+    }, [isEditFlow, apiClient.listId, item]);
```

Comment:
Why did you need to add this?

It should have been taken care of by 209 when `item` is provided on input to this component (ex. user clicked edit on the card list).

Reply:
That sounds like a bug. The intent is for the form to be provided the `item` from the list so you should not have the need to do this. I'll do a little more investigation on that.